js/src/classic-editor.js

```diff
@@ -216,6 +203,29 @@ jQuery(
 			}
 		);
 
+		// Listen to `onPostLangChoice` to perform actions after the language has been changed.
+		document.addEventListener(
+			'onPostLangChoice',
+			() => {
+				// Update the old language with the new one to be able to compare it in the next changing.
+				initializeLanguageOldValue();
+				// modifies the language in the tag cloud
+				$( '.tagcloud-link' ).each(
+					function () {
+						var id = $( this ).attr( 'id' );
+						tagBox.get( id );
+					}
+				);
+
+				// Modifies the text direction
+				$( 'body' ).removeClass( 'pll-dir-rtl' ).removeClass( 'pll-dir-ltr' ).addClass( 'pll-dir-' + dir );
+				$( '#content_ifr' ).contents().find( 'html' ).attr( 'lang', lang ).attr( 'dir', dir );
+				$( '#content_ifr' ).contents().find( 'body' ).attr( 'dir', dir );
+
+				pll.media.resetAllAttachmentsCollections();
```

Comment:
Could all this data be passed in the custom event?
Using the CustomEvent API ([see](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent)).